File: backend/app/schemas/organization_api_keys.py
"""Schemas for organization LLM API key management."""

from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, Field


class OrganizationApiKeyProviderStatus(BaseModel):
    provider: str
    label: str
    configured: bool
    is_active: bool = False
    masked_key: str | None = None
    updated_at: datetime | None = None


class OrganizationApiKeyListResponse(BaseModel):
    items: list[OrganizationApiKeyProviderStatus]


class OrganizationApiKeyUpsertRequest(BaseModel):
    provider: str = Field(min_length=2, max_length=32)
    api_key: str = Field(min_length=8, max_length=512)
    is_active: bool = True


class OrganizationApiKeyUpsertResponse(BaseModel):
    provider: str
    configured: bool
    masked_key: str | None
    is_active: bool
    message: str = "API key saved."

File: frontend/src/types/organization-api-keys.ts
export interface OrganizationApiKeyProviderStatus {
  provider: string;
  label: string;
  configured: boolean;
  is_active: boolean;
  masked_key: string | null;
  updated_at: string | null;
}

export interface OrganizationApiKeyListResponse {
  items: OrganizationApiKeyProviderStatus[];
}

export interface OrganizationApiKeyUpsertRequest {
  provider: string;
  api_key: string;
  is_active?: boolean;
}

export interface OrganizationApiKeyUpsertResponse {
  provider: string;
  configured: boolean;
  masked_key: string | null;
  is_active: boolean;
  message: string;
}

export const LLM_API_KEY_PROVIDERS = [
  { id: "openai", label: "OpenAI" },
  { id: "anthropic", label: "Anthropic" },
  { id: "deepseek", label: "DeepSeek" },
  { id: "gemini", label: "Google Gemini" },
] as const;
